src/ops.jl

```diff
@@ -356,3 +356,43 @@ Base.floor{T}(::Type{T}, A::FixedArray) = map(x->floor(T, x), A)
 Base.ceil{T}( ::Type{T}, A::FixedArray) = map(x->ceil( T, x), A)
 Base.trunc{T}(::Type{T}, A::FixedArray) = map(x->trunc(T, x), A)
 Base.round{T}(::Type{T}, A::FixedArray) = map(x->round(T, x), A)
+
+import Base.Cartesian.@nif
+shift(t::Tuple) = Base.tail(t)
+unshift(t::Tuple, val) = tuple(val, t...)
+push(t::Tuple, val) = tuple(t..., val)
+@generated unpush{N,T}(t::NTuple{N,T}) = begin # is there a better name?
```

Comment:
Yeah the `v[2:end]` thing will not be type-stable. I am happy with renaming shift to tail if there is also a standard name for pop. Otherwise I prefer shift+unshift/push+pop to some more obscure naming scheme.